### DISCUSSION TITLE
Plotting using logarithmic scale for y-axis

### DISCUSSION THREAD

**Discussion by s1291** [saulpw/visidata#1394] [Q&A]
Hi,
I have been using Visidata lately for some plots and I've to admit that it is very interesting.

I need to plot some data with a logarithmic scale for the y-axis. But I can't find any relevant documentation about it.

Is it possible to achieve that?

Thank you

**Comment by saulpw** [comment]
Hi @s1291, you could make a log column on the source sheet with e.g. `=log(y)`, and then plot that.  Does that get you what you want?

**Reply by s1291** [reply]
Thank you for your reply.

I have multiple columns (about 15 columns) and creating `log(yi)` column for each column `yi` will increase the size of the sheet. I can't also get the formatting on the y-axis to be 10^0, 10^-1, 10^-2, etc.

So probably it would be much easier if Visidata provides the feature natively to allow toggling between "linear" and {"log-y", "log-x", or both "log-x" and "log-y"} modes.

Thanks

**Comment by andycraig** [comment]
I'd really like this feature as well and have been thinking about trying to create a plugin to do it.

I think what the plugin would need to do is patch `Canvas` and `InvertedCanvas` with new versions that allow each axis to be in either 'linear' or 'log' mode. (Then possibly patch `GraphSheet` too to inherit from the patched `InvertedCanvas`?) https://github.com/saulpw/visidata/blob/924563a0b7eeede80834dc02d8c1f237fe1949c1/visidata/graph.py#L11

Not sure when I'll get to this but it's on my list.
